src/overlays/actors/ovl_En_Wiz_Fire/z_en_wiz_fire.c

```diff
@@ -378,231 +407,228 @@ void func_80A4A11C(EnWizFire* this, PlayState* play) {
+            sPoolHitByIceArrow = false;
             Actor_MarkForDeath(&this->actor);
         }
+
         return;
     }
 
-    Math_ApproachZeroF(&this->unk_1F4, 0.1f, 0.01f);
+    Math_ApproachZeroF(&this->bigFlameScale, 0.1f, 0.01f);
 
-    if (this->unk_1F4 < 0.01f) {
-        Math_ApproachZeroF(&this->unk_1FC, 1.0f, 10.0f);
-        if ((this->unk_1FC < 10.0f) && (this->unk_204 < 0.001f)) {
-            D_80A4C1C0 = 0;
-            if ((this->actor.parent != NULL) && (this->unk_162 == 0) && (this->actor.parent->id == ACTOR_EN_WIZ) &&
-                (this->actor.parent->update != NULL)) {
+    if (this->bigFlameScale < 0.01f) {
+        Math_ApproachZeroF(&this->alpha, 1.0f, 10.0f);
+        if ((this->alpha < 10.0f) && (this->blendScaleFrac < 0.001f)) {
+            sPoolHitByIceArrow = false;
+            if ((this->actor.parent != NULL) && (this->type == EN_WIZ_FIRE_TYPE_MAGIC_PROJECTILE) &&
+                (this->actor.parent->id == ACTOR_EN_WIZ) && (this->actor.parent->update != NULL)) {
                 EnWiz* wiz = (EnWiz*)this->actor.parent;
 
-                wiz->unk_3BA = 0;
+                wiz->hasActiveProjectile = false;
             }
+
             Actor_MarkForDeath(&this->actor);
         }
     }
 }
 
-void func_80A4A608(EnWizFire* this, PlayState* play) {
-    if (this->unk_16A == 0) {
-        this->unk_16A = 2;
-        this->unk_16E++;
-        if (this->unk_16E >= 6) {
-            if ((this->actor.parent != NULL) && (this->unk_162 == 0) && (this->actor.parent->id == ACTOR_EN_WIZ)) {
+void EnWiz_KillMagicProjectile(EnWizFire* this, PlayState* play) {
+    if (this->increaseLowestValidMagicProjectileIndexTimer == 0) {
+        this->increaseLowestValidMagicProjectileIndexTimer = 2;
+        this->lowestValidMagicProjectileIndex++;
+        if (this->lowestValidMagicProjectileIndex >= 6) {
+            if ((this->actor.parent != NULL) && (this->type == EN_WIZ_FIRE_TYPE_MAGIC_PROJECTILE) &&
+                (this->actor.parent->id == ACTOR_EN_WIZ)) {
                 EnWiz* wiz = (EnWiz*)this->actor.parent;
 
-                D_80A4C1C0 = 0;
+                sPoolHitByIceArrow = false;
                 if (wiz->actor.update != NULL) {
-                    wiz->unk_3BA = 0;
+                    wiz->hasActiveProjectile = false;
                 }
             }
+
             Actor_MarkForDeath(&this->actor);
         }
     }
 }
 
 void EnWizFire_Update(Actor* thisx, PlayState* play2) {
-    static Color_RGB8 D_80A4C234[] = {
-        { 100, 40, 40 }, { 180, 0x78, 80 }, { 155, 80, 80 },   { 125, 20, 0 },
-        { 0, 0, 0 },     { 200, 250, 250 }, { 100, 250, 250 }, { 225, 255, 235 },
+    // These are AmbientColor, DiffuseColor1, DiffuseColor2, and fogColor
+    static Color_RGB8 lightSettingsColors[] = {
+        { 100, 40, 40 }, { 180, 120, 80 },  { 155, 80, 80 },   { 125, 20, 0 },    // Fire
+        { 0, 0, 0 },     { 200, 250, 250 }, { 100, 250, 250 }, { 225, 255, 235 }, // Ice
     };
```

Comment:
Can't be a 2D array, right?

Reply:
I'm almost certain it can't. This line in particular:
```c
index = this->isIceType * 4;
```
Is very fussy, and the fact that `index` is an `s16` here is very important. I can't think of a way to write this as a 2D array without introducing some serious jank.